server/datastore/mysql/locks.go

```diff
@@ -56,3 +57,27 @@ func (d *Datastore) Unlock(ctx context.Context, name string, owner string) error
 	_, err := d.writer.ExecContext(ctx, `DELETE FROM locks WHERE name = ? and owner = ?`, name, owner)
 	return err
 }
+
+func (d *Datastore) DBLocks(ctx context.Context) ([]*fleet.DBLock, error) {
+	stmt := `
+    SELECT
+      r.trx_id              waiting_trx_id,
+      r.trx_mysql_thread_id waiting_thread,
+      r.trx_query           waiting_query,
+      b.trx_id              blocking_trx_id,
+      b.trx_mysql_thread_id blocking_thread,
+      b.trx_query           blocking_query
+    FROM       information_schema.innodb_lock_waits w
+    INNER JOIN information_schema.innodb_trx b
+      ON b.trx_id = w.blocking_trx_id
+    INNER JOIN information_schema.innodb_trx r
+      ON r.trx_id = w.requesting_trx_id`
```

Comment:
There are some issues with this query, although it works fine in tests:

* It requires the `PROCESS` privilege, which the mysql configured user may not  have (i.e. when manually testing this locally with fleetctl against the running fleet instance it failed with `Error 1227: Access denied; you need (at least one of) the PROCESS privilege(s) for this operation`.
* The `INNODB_LOCK_WAITS` is deprecated and removed in mysql 8.0: https://dev.mysql.com/doc/refman/5.7/en//information-schema-innodb-lock-waits-table.html, replaced with `performance_schema.data_lock_waits` which also has privilege requirements.

Should we punt on the `db-locks` sub-command or do we want to support this (we can detect if the table exists or not, or the mysql version, and build the query accordingly). This means we'd need to make it clear in the docs that a user with sufficient privileges is required (no big deal if that's not the case except that db-locks will fail - the command could add a message like "make sure the mysql user has sufficient privilege" and link to the docs).

Reply:
Let's make sure it works for plain MySQL 5.7, and if it fails, just make sure it doesn't prevent other things from working. We can expand support later on. Most paid customers are in 5.7, although some are in Percona's version.